tests/faction_camp_test.cpp

```diff
@@ -0,0 +1,60 @@
+#include <cstddef>
+#include <iosfwd>
+#include <utility>
+#include <vector>
+
+#include "basecamp.h"
+#include "cata_catch.h"
+#include "clzones.h"
+#include "coordinates.h"
+#include "faction.h"
+#include "map.h"
+#include "map_helpers.h"
+#include "overmap.h"
+#include "overmapbuffer.h"
+#include "player_helpers.h"
+
+static const zone_type_id zone_type_CAMP_FOOD( "CAMP_FOOD" );
+static const zone_type_id zone_type_CAMP_STORAGE( "CAMP_STORAGE" );
+
+TEST_CASE( "camp_calorie_counting", "[camp]" )
+{
+    clear_map();
+    map &m = get_map();
+    tripoint const zone_loc = m.getabs( tripoint{ 5, 5, 0 } );
+    mapgen_place_zone( zone_loc, zone_loc, zone_type_CAMP_FOOD, your_fac, {},
+                       "food" );
+    mapgen_place_zone( zone_loc, zone_loc, zone_type_CAMP_STORAGE, your_fac, {},
+                       "storage" );
+    get_player_character().setpos( zone_loc );
```

Comment:
I think this could cause a map shift, meaning the `zone_loc` used below is in the wrong place. You should get the global coords before doing this.